Description: Install dict to versioned dir
Bug-Debian: https://bugs.debian.org/709123
Forwarded: no
Author: Mathieu Malaterre <malat@debian.org>
Index: dcmtk/CMake/GenerateDCMTKConfigure.cmake
===================================================================
--- dcmtk.orig/CMake/GenerateDCMTKConfigure.cmake
+++ dcmtk/CMake/GenerateDCMTKConfigure.cmake
@@ -114,15 +114,7 @@ endif()
 #endif()
 
 # Configure file
-
-# Windows being windows, it lies about its processor type to 32 bit binaries
-set(SYSTEM_PROCESSOR "$ENV{PROCESSOR_ARCHITEW6432}")
-if(NOT SYSTEM_PROCESSOR)
-  set(SYSTEM_PROCESSOR "${CMAKE_SYSTEM_PROCESSOR}")
-endif()
-# CMake doesn't provide a configure-style system type string
-set(CANONICAL_HOST_TYPE "${SYSTEM_PROCESSOR}-${CMAKE_SYSTEM_NAME}")
-DCMTK_UNSET(SYSTEM_PROCESSOR)
+set(CANONICAL_HOST_TYPE "Debian")
 
 # Configure dictionary path and install prefix
 if(WIN32 AND NOT CYGWIN)
@@ -134,10 +126,10 @@ if(WIN32 AND NOT CYGWIN)
   set(ENVIRONMENT_PATH_SEPARATOR ";")
   # Set dictionary path to the data dir inside install main dir (prefix)
   if(DCMTK_ENABLE_EXTERNAL_DICTIONARY)
-    set(DCM_DICT_DEFAULT_PATH "${DCMTK_PREFIX}\\\\${CMAKE_INSTALL_DATADIR}\\\\dcmtk\\\\dicom.dic")
+    set(DCM_DICT_DEFAULT_PATH "${DCMTK_PREFIX}\\\\${CMAKE_INSTALL_DATADIC}\\\\dicom.dic")
     # If private dictionary should be utilized, add it to default dictionary path.
     if(ENABLE_PRIVATE_TAGS)
-      set(DCM_DICT_DEFAULT_PATH "${DCM_DICT_DEFAULT_PATH};${DCMTK_PREFIX}\\\\${CMAKE_INSTALL_DATADIR}\\\\dcmtk\\\\private.dic")
+      set(DCM_DICT_DEFAULT_PATH "${DCM_DICT_DEFAULT_PATH};${DCMTK_PREFIX}\\\\${CMAKE_INSTALL_DATADIC}\\\\private.dic")
     endif()
      # Again, for Windows strip all / from path and replace it with \\.
     string(REGEX REPLACE "/" "\\\\\\\\" DCM_DICT_DEFAULT_PATH "${DCM_DICT_DEFAULT_PATH}")
@@ -155,10 +147,10 @@ else()
   set(ENVIRONMENT_PATH_SEPARATOR ":")
   # Set dictionary path to the data dir inside install main dir (prefix).
   if(DCMTK_ENABLE_EXTERNAL_DICTIONARY)
-    set(DCM_DICT_DEFAULT_PATH "${DCMTK_PREFIX}/${CMAKE_INSTALL_DATADIR}/dcmtk/dicom.dic")
+    set(DCM_DICT_DEFAULT_PATH "${DCMTK_PREFIX}/${CMAKE_INSTALL_DATADIC}/dicom.dic")
     # If private dictionary should be utilized, add it to default dictionary path.
     if(ENABLE_PRIVATE_TAGS)
-      set(DCM_DICT_DEFAULT_PATH "${DCM_DICT_DEFAULT_PATH}:${DCMTK_PREFIX}/${CMAKE_INSTALL_DATADIR}/dcmtk/private.dic")
+      set(DCM_DICT_DEFAULT_PATH "${DCM_DICT_DEFAULT_PATH}:${DCMTK_PREFIX}/${CMAKE_INSTALL_DATADIC}/private.dic")
     endif()
   else()
     set(DCM_DICT_DEFAULT_PATH "")
Index: dcmtk/dcmdata/data/CMakeLists.txt
===================================================================
--- dcmtk.orig/dcmdata/data/CMakeLists.txt
+++ dcmtk/dcmdata/data/CMakeLists.txt
@@ -1,5 +1,8 @@
 # declare installation files
-install(FILES dicom.dic acrnema.dic private.dic diconde.dic dcm2xml.dtd dumppat.txt SC.dump VLP.dump DESTINATION "${CMAKE_INSTALL_DATADIR}/dcmtk" COMPONENT data)
+set(DCMTK_INSTALL_DATDIR "share/libdcmtk${DCMTK_ABI_VERSION}")
+
+install(FILES dcm2xml.dtd dumppat.txt SC.dump VLP.dump DESTINATION "${CMAKE_INSTALL_DATADIR}/dcmtk" COMPONENT data)
+install(FILES dicom.dic acrnema.dic private.dic diconde.dic DESTINATION ${CMAKE_INSTALL_DATADIC} COMPONENT data)
 
 # add paths to the dictionaries to a CMake variable for being used by the unit tests
 set(DCMTK_DICOM_DICTIONARIES
